### ISSUE TITLE
Audit behavioral differences between CoreFX and Desktop

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#15131]
Eg by looking through all tests that are desktop specific by grepping for
`SkipOnTargetFramework.*TargetFrameworkMonikers.Netcoreapp`

We are concerned about cases where we broke compat with Desktop exception behavior, and now with the new focus on NS2.0 we want to be more closely compatible. See https://github.com/dotnet/corefx/pull/15095/files/51dfc8f3bb7a08f87ba519ed39ec70a443b35507#diff-e33ac35956193577452d988d644a7866

**Comment by Clockwork-Muse** [comment]
As a reminder to whoever grabs this, there are some tests that aren't skipped, but rather just choose a different internal path, possibly by `RuntimeDetection.IsNetFramework` or similar.

**Comment by danmosemsft** [comment]
@sepidehMS I think you did this or do I misremember? I guess you went through the tests, but nobody looked through the implementation?
If so that's probably enough for you to close this.
@safern

**Comment by sepidehMS** [comment]
@danmosemsft I did go through tests that were skipped in a way on some platform(s) and analyzed if the skipping makes sense and has a valid reason (https://github.com/dotnet/corefx/pull/15932). I did not go through the implementation or those that take different internal paths, though.

**Comment by sepidehMS** [comment]
I think I might have caused a bit of confusion. My pass went through only **PlatformSpecific** tests (e.g. `PlatformSpecific(TestPlatforms.Windows)`) and did not include investigating differences between Desktop and CoreFX.

**Comment by safern** [comment]
@danmosemsft with the Desktop runs in Jenkins and Helix we will be able to identify this differences in behavior while opening test failures issues.

**Comment by danmosemsft** [comment]
@sepidehMS OK
@safern if I understand right you aren't checking that desktop specific tests have a good reason for existing, only that they pass. Ie., you won't audit `SkipOnTargetFramework.*TargetFrameworkMonikers.Netcoreapp`. Which is fine (wasn't expecting you to) but seems we could leave this open.

**Comment by safern** [comment]
Yes that is correct. Yes it sound reasonable to leave this open to audit those cases.

**Comment by ViktorHofer** [comment]
@danmosemsft I think this can be closed now?
